transport/internal/s2a/s2a.go

```diff
@@ -0,0 +1,12 @@
+package s2autil
+
+func MtlsEndpointIsEnabled(mtlsEndpoint string) bool {
```

Comment:
So how do you plan to maintain the source of truth for such an allowlist? Is the list specific to s2a or for all mTLS GCP apis in general? Note that the existing mTLS implementations in this library does not rely on such an allowlist - furthermore, the official mTLS endpoints used by this library are generated through what is known as the "discovery doc" tool chain

Reply:
Thanks Andy, this was after chatting with Shin, we learned that even though the mtls endpoints are set in client libraries, sometimes it doesn't necessarily work. 
We are currently working with the team behind the `discovery` API to add an extra indicator (for each service) to mark whether an mtls endpoint is indeed MTLS enabled.  The plan is the default value will be `false` and only when we've confirmed with the service owner and verified it, it'll be set to true.

This function should mostly return that value from discovery doc. It can potentially take in a config (either from file or online) for overriding purpose. 

I'll share more details as we make progress on this. Plz let us know if you have any thoughts/suggestion on this.